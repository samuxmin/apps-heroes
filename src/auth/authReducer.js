import { types } from "../types/types";

// const state = {
//     name:'Samuel',
//     logged:false
// }

export const authReducer = (state = {}, action) => {
   
    switch (action.type) {
        case types.login:
            return{
                ...action.payload, 
                logged:true
            }
        case types.logout:
            return{
                logged:false
            }

        default:
            return state;
    }
}
