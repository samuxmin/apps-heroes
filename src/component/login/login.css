.login-input{
    height: 30px;
    color: black;
    margin: 10px;
}