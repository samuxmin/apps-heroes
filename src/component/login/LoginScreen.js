import React, { useContext } from "react";
import { AuthContext } from "../../auth/AuthContext";
import { useForm } from "../../hooks/useForm";
import { types } from "../../types/types";
import "./login.css";

export const LoginScreen = ({ history }) => {
  const { dispatch } = useContext(AuthContext);
  const [{ username, password }, handleInputChange] = useForm({
    username: "",
    password: "",
  });
  const action = {
    payload: { name: username },
    type: types.login,
  };
  
  const handleLogin = () => {
    const lastPath = localStorage.getItem("lastPath") || "/";
    dispatch(action);
    history.replace(lastPath);
  };

  return (
    <div className="app">
      <h1>Login Screen</h1>
      <hr />
      <label htmlFor="username">
        <b>Username</b>
      </label>
      <input
        className="login-input"
        type="text"
        placeholder="Username here..."
        name="username"
        autoComplete="off"
        value={username}
        onChange={handleInputChange}
      />
      <br />
      <label htmlFor="password">
        <b>Password</b>
      </label>
      <input
        className="login-input"
        type="password"
        placeholder="Password here..."
        name="password"
        value={password}
        onChange={handleInputChange}
      />
      <p>(Cualquier contrase√±a es valida)</p>
      <button className="btn btn-login" onClick={handleLogin}>
        Login
      </button>
    </div>
  );
};
