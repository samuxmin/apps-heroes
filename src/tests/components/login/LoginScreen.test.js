import React from "react";
import { mount } from "enzyme";
import { LoginScreen } from "../../../component/login/LoginScreen";
import { AuthContext } from "../../../auth/AuthContext";
import { types } from "../../../types/types";

describe("Pruebas en <LoginScreen />", () => {
  const historyMock = {
    replace: jest.fn(),
  };
  const dispatch = jest.fn();

  const wrapper = mount(
    <AuthContext.Provider value={{ dispatch }}>
      <LoginScreen history={historyMock} />
    </AuthContext.Provider>
  );
  test("debe mostrarse correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("debe llamar al dispatch y history.replace", () => {
    const handleClick = wrapper.find("button").prop('onClick');
    handleClick()

    expect(dispatch).toHaveBeenCalledWith({
        payload: { name: "" },
        type: types.login,
      });

      expect(historyMock.replace).toHaveBeenCalledWith('/');
      localStorage.setItem( 'lastPath', '/awita');
      handleClick();

      expect(historyMock.replace).toHaveBeenCalledWith('/awita')
  });
});
