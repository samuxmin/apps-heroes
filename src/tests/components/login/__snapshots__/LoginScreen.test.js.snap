// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <LoginScreen /> debe mostrarse correctamente 1`] = `
<div
  className="app"
>
  <h1>
    Login Screen
  </h1>
  <hr />
  <label
    htmlFor="username"
  >
    <b>
      Username
    </b>
  </label>
  <input
    autoComplete="off"
    className="login-input"
    name="username"
    onChange={[Function]}
    placeholder="Username here..."
    type="text"
    value=""
  />
  <br />
  <label
    htmlFor="password"
  >
    <b>
      Password
    </b>
  </label>
  <input
    className="login-input"
    name="password"
    onChange={[Function]}
    placeholder="Password here..."
    type="password"
    value=""
  />
  <p>
    (Cualquier contrase√±a es valida)
  </p>
  <button
    className="btn btn-login"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
